sql/catalog/lease: TestLeaseRenewedPeriodically failed

sql/catalog/lease.TestLeaseRenewedPeriodically [failed](https://tanzanite.cluster.engflow.com/invocations/default/3b2c11df-80d8-499b-be08-33fb74892600?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on release-24.2 @ [9d62e396e2d143a7b718e0279981f5f0001cccd8](https://github.com/cockroachdb/cockroach/commits/9d62e396e2d143a7b718e0279981f5f0001cccd8):


```
=== RUN   TestLeaseRenewedPeriodically
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseRenewedPeriodically3884335860
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster.go:482: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_COMMIT_DEADLINE_EXCEEDED - txn timestamp pushed too much; deadline exceeded by 570.834994ms (1736766049.766686878,2 > 1736766049.195851884,0)): "sql txn" meta={id=0794d64a key=/Table/63/1/0/0 iso=Serializable pri=0.03448646 epo=4 ts=1736766049.766686878,2 min=1736766037.220369605,0 seq=130} lock=true stat=PENDING rts=1736766049.766686878,2 wto=false gul=1736766037.720369605,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseRenewedPeriodically3884335860
--- FAIL: TestLeaseRenewedPeriodically (135.44s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseRenewedPeriodically.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46427

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -1894,6 +1894,8 @@ func TestLeaseRenewedPeriodically(testingT *testing.T) {
 	defer leaktest.AfterTest(testingT)()
 	defer log.Scope(testingT).Close(testingT)
 
+	skip.UnderRace(testingT)
+
 	ctx := context.Background()
 
 	var mu syncutil.Mutex

